include/boost/math/special_functions/jacobi_theta.hpp

```diff
@@ -0,0 +1,646 @@
+// [20.7.31] sqrt(-iτ)θ₂(z|τ) =    exp(iτ'z²/π)*θ₄(zτ'|τ')
+// [20.7.32] sqrt(-iτ)θ₃(z|τ) =    exp(iτ'z²/π)*θ₃(zτ'|τ')
+// [20.7.33] sqrt(-iτ)θ₄(z|τ) =    exp(iτ'z²/π)*θ₂(zτ'|τ')
+template <class RealType, class Policy>
+inline RealType
+_IMAGINARY_jacobi_theta1tau(RealType z, RealType tau, const Policy& pol) {
+    BOOST_MATH_STD_USING
+    unsigned n = 0;
+    RealType eps = policies::get_epsilon<RealType, Policy>();
+    RealType /* q_n, */ delta, delta1, delta2, result = RealType(0);
+
+    do {
+        delta1 = exp(tau*(z*(RealType(2*n+1) - z/constants::pi<RealType>())
+                    -constants::pi<RealType>() * RealType(n + 0.5)*RealType(n + 0.5)));
+        // q_n = exp(-tau * constants::pi<RealType>() * RealType(n + 0.5)*RealType(n + 0.5) );
+        delta2 = exp(tau*(z*(-RealType(2*n+1) - z/constants::pi<RealType>())
+                    -constants::pi<RealType>() * RealType(n + 0.5)*RealType(n + 0.5)));
+
+        if (n%2) {
+            delta = delta2 - delta1;
+        } else {
+            delta = delta1 - delta2;
+        }
+
+        result += delta;
+        n++;
+    } while (abs(delta * sqrt(tau)) > eps || (abs(result * sqrt(tau)) > eps * eps && abs(delta/result) > eps));
+    
+    if (abs(result) < eps * eps)
+        return RealType(0);
+
+    return result * sqrt(tau);
+}
+
+template <class RealType, class Policy>
+inline RealType
+_IMAGINARY_jacobi_theta2tau(RealType z, RealType tau, const Policy& pol) {
+    BOOST_MATH_STD_USING
+    unsigned n = 0;
+    RealType eps = policies::get_epsilon<RealType, Policy>();
+    RealType /* q_n, */ delta, delta1, delta2, result = RealType(0);
+
+    do {
+        delta1 = exp(tau*(z*(RealType(2*n+1) - z/constants::pi<RealType>())
+                    -constants::pi<RealType>() * RealType(n + 0.5)*RealType(n + 0.5)));
+        // q_n = exp(-tau * constants::pi<RealType>() * RealType(n + 0.5)*RealType(n + 0.5) );
+        delta2 = exp(tau*(z*(-RealType(2*n+1) - z/constants::pi<RealType>())
+                    -constants::pi<RealType>() * RealType(n + 0.5)*RealType(n + 0.5)));
+
+        delta = delta1 + delta2;
+
+        result += delta;
+        n++;
+    } while (abs(delta * sqrt(tau)) > eps || (abs(result * sqrt(tau)) > eps * eps && abs(delta/result) > eps));
+    
+    if (abs(result) < eps * eps)
+        return RealType(0);
+
+    return result * sqrt(tau);
+}
+
+template <class RealType, class Policy>
+inline RealType
+_IMAGINARY_jacobi_theta3tau(RealType z, RealType tau, const Policy& pol) {
+    BOOST_MATH_STD_USING
+    unsigned n = 1;
+    RealType eps = policies::get_epsilon<RealType, Policy>();
+    RealType /* q_n, */ delta, delta1, delta2, result = exp(-z*z*tau/constants::pi<RealType>());
+
+    do {
+        delta1 = exp(tau*(z*(RealType(2*n) - z/constants::pi<RealType>())
+                    -constants::pi<RealType>() * RealType(n)*RealType(n)));
+        // q_n = exp(-tau * constants::pi<RealType>() * RealType(n + 0.5)*RealType(n + 0.5) );
```

Comment:
This is commented out in every implementation.

Reply:
Removed in latest commit.